codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/Ec2Query.java

```diff
@@ -34,9 +34,21 @@ protected void generateDocumentBodyShapeDeserializers(GenerationContext context,
     @Override
     protected void writeErrorMessageCodeDeserializer(GenerationContext context) {
         GoWriter writer = context.getWriter();
+        writer.write("m := middleware.Metadata{}");
         writer.addUseImports(AwsGoDependency.AWS_EC2QUERY_PROTOCOL);
-        writer.write("errorCode, errorMessage, err := ec2query.GetResponseErrorCode(errorBody)");
-        writer.write("if err != nil { return err }");
+        writer.write("errorComponents, err := ec2query.GetErrorResponseComponents(errorBody)");
+        writer.write("if err != nil { return m, err }");
+        writer.insertTrailingNewline();
+
+        writer.addUseImports(AwsGoDependency.AWS_HTTP_TRANSPORT);
+        writer.write("awshttp.SetRequestIDMetadata(&m, errorComponents.RequestID)");
+        writer.insertTrailingNewline();
+
+        // assign middleware metadata to metadata var
```

Comment:
Why duplicate the `metadata` parameter?  Won't this risk deleting any other metadata that may of been stored on the `metadata` from further down the stack?